shared/js/background.js

```diff
@@ -242,21 +246,44 @@ chrome.webRequest.onBeforeRequest.addListener(
             return
         }
 
+        if (thisTab.failedUpgradeUrls[requestData.url]) {
+            console.log('already tried https upgrades for this URL and failed, skip:\n' + requestData.url)
+            return
+        }
```

Comment:
`failedUpgradeUrls` serves as a temporary whitelist that the tab builds up as it encounters failing URLs. Maybe we can share this between all tabs by storing it in `https.js`.